cloud/analytics.go

```diff
@@ -2,53 +2,15 @@ package cloud
 
 import (
 	"context"
-	"encoding/json"
-	"net/http"
 
+	"github.com/earthly/cloud-api/analytics"
 	"github.com/pkg/errors"
 )
 
-// EarthlyAnalytics is the payload used in SendAnalytics.
-// It contains information about the command that was run,
-// the environment it was run in, and the result of the command.
-type EarthlyAnalytics struct {
-	Key              string                    `json:"key"`
-	InstallID        string                    `json:"install_id"`
-	Version          string                    `json:"version"`
-	Platform         string                    `json:"platform"`
-	BuildkitPlatform string                    `json:"buildkit_platform"`
-	UserPlatform     string                    `json:"user_platform"`
-	GitSHA           string                    `json:"git_sha"`
-	ExitCode         int                       `json:"exit_code"`
-	CI               string                    `json:"ci_name"`
-	IsSatellite      bool                      `json:"is_satellite"`
-	SatelliteVersion string                    `json:"satellite_version"`
-	IsRemoteBuildkit bool                      `json:"is_remote_buildkit"`
-	RepoHash         string                    `json:"repo_hash"`
-	ExecutionSeconds float64                   `json:"execution_seconds"`
-	Terminal         bool                      `json:"terminal"`
-	Counts           map[string]map[string]int `json:"counts"`
-}
-
 // SendAnalytics send an analytics event to the Cloud server.
-func (c *client) SendAnalytics(ctx context.Context, data *EarthlyAnalytics) error {
-	payload, err := json.Marshal(data)
-	if err != nil {
-		return errors.Wrap(err, "failed to marshal data")
-	}
-	opts := []requestOpt{
-		withBody(payload),
-		withHeader("Content-Type", "application/json; charset=utf-8"),
-	}
-	if c.IsLoggedIn(ctx) {
-		opts = append(opts, withAuth())
```

Comment:
Have you verified that we still track if a user is logged in or not?

Reply:
yes, but I plan to do another round of testing when this PR seems close to done.

Reply:
perfect! just wanted to double check.

Reply:
so, good call!
of course I was planning on testing, and glad I caught it, but when I originally tested I used my own client and I overlooked the fact that earthly uses a client interceptor to authenticate, so I had to address that.